src/components/application_manager/src/app_service_manager.cc

```diff
@@ -106,4 +149,91 @@ std::vector<smart_objects::SmartObject> AppServiceManager::GetAllServices() {
   return services;
 }
 
+void AppServiceManager::setServicePublished(const std::string service_id,
+                                            bool service_published) {
+  auto it = published_services_.find(service_id);
+  if (it == published_services_.end()) {
+    LOG4CXX_ERROR(logger_, "Service id does not exist in published services");
+    return;
+  }
+  it->second.record[strings::service_published] = service_published;
+}
+
+void AppServiceManager::setServiceActive(const std::string service_id,
+                                         bool service_activated) {
+  auto it = published_services_.find(service_id);
+  if (it == published_services_.end()) {
+    LOG4CXX_ERROR(logger_, "Service id does not exist in published services");
+    return;
+  }
+  it->second.record[strings::service_active] = service_activated;
+}
+
+smart_objects::SmartObject CreateMobileSystemCapabilityNotification(
+    AppServiceManager& app_service_manager,
+    const std::string service_id,
+    mobile_apis::ServiceUpdateReason::eType update_reason) {
+  smart_objects::SmartObject message(smart_objects::SmartType_Map);
+
+  message[strings::params][strings::message_type] = MessageType::kNotification;
+  message[strings::params][strings::function_id] =
+      mobile_apis::FunctionID::OnSystemCapabilityUpdatedID;
+
+  smart_objects::SmartObject system_capability =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  system_capability[strings::system_capability_type] =
+      mobile_apis::SystemCapabilityType::APP_SERVICES;
```

Comment:
Future Note: There might be other types of capabilities that can be updated other than app services. For example publishing a nav service could update the NAVIGATION system capability type in addition to the APP_SERVICE type. Lets confirm this in the evolution process during the NAVIGATION type review. Hold off on making the change for now.